src/object/obj_rpc.c

```diff
@@ -295,25 +295,19 @@ static int crt_proc_daos_iom_t(crt_proc_t proc, daos_iom_t *map)
 	if (iom_nr == 0)
 		return 0;
 
-	switch (proc_op) {
-	case CRT_PROC_DECODE:
+	if (DECODING(proc_op)) {
 		map->iom_nr = iom_nr;
 		map->iom_nr_out = iom_nr;
 		D_ALLOC_ARRAY(map->iom_recxs, iom_nr);
 		if (map->iom_recxs == NULL)
 			return -DER_NOMEM;
-		/* fall through to fill iom_recxs */
-	case CRT_PROC_ENCODE:
-		rc = crt_proc_memcpy(proc, map->iom_recxs,
-				     iom_nr * sizeof(*map->iom_recxs));
-		if (unlikely(rc)) {
-			if (DECODING(proc_op))
```

Comment:
I don't understand this code before, is this encoding or decoding?

Reply:
It's used for both. It switch the src and dst in case of encode/decode.